src/main/middleware/Authentication.ts

```diff
@@ -0,0 +1,35 @@
+import { NextFunction } from 'express';
+import { Request, Response } from 'express';
+import { auth } from 'firebase-admin/lib/auth';
+import Auth = auth.Auth;
+
+/**
+ * Middleware function to check if the request has correct client header using firebase-admin auth
+ */
+export class FirebaseAuth {
```

Comment:
I noticed the filename is `Authentication` whereas the class is `FirebaseAuth`. Is there a reason for making the two different?

Reply:
No, I refactored!